app/controllers/sessions_controller.rb

```diff
@@ -9,6 +9,10 @@ def create
 
     if @user
       session[:user_id] = @user.id
+      @auth = request.env['omniauth.auth']
+      @token = @auth["credentials"]["token"]
+      ENV['token'] = @token
```

Comment:
how come you were putting @token into the Environment? It looks like it is only being used in the show.html.erb view and wouldn't @token have been enough to display it there?